include/deepworks/optimizers.md

```diff
@@ -0,0 +1,65 @@
+# Optimization algorithms
+### Stochastic gradient descent
+Implements stochastic gradient descent.
+#### Attributes
+- `lr` – step size
+#### Example
+```cpp
+float lr = 1e-3f;
+dw::optimizer::SGD opt(model.params(), lr);
+
+...
+
+loss = criterion.forward(predict, y);
+criterion.backward(predict, y, grad_output);
+model.backward(X, predict, grad_output);
+
+opt.step();
+```
+
+### Stochastic gradient descent with momentum
+Implements stochastic gradient descent with momentum. It is a technique for accelerating gradient descent that accumulates a velocity
+vector in directions of persistent reduction in the objective across iterations.
+#### Attributes
+- `lr` – step size
+- `gamma` – exponential decay rate for the velocity
+vector (default: 0.9)
+#### Example
+```cpp
+float lr = 1e-3f;
+float gamma = 0.95f;
+dw::optimizer::SGDMomentum opt(model.params(), lr, gamma);
+// dw::optimizer::SGDMomentum opt(model.params(), lr);
```

Comment:
Why it's commented ?

Reply:
i want to show, that user can pass all the parameters, or use the default ones and leave only model.params and learning rate

Reply:
A bit confusing, but OK let it be